site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx

```diff
@@ -9,71 +9,104 @@ import duration from "dayjs/plugin/duration"
 
 dayjs.extend(duration)
 
+// ActiveTransition gets the build estimate for the workspace,
+// if it is in a transition state.
+export const ActiveTransition = (
+  template: Template,
+  workspace: Workspace,
+): TransitionStats | undefined => {
+  const status = workspace.latest_build.status
+
+  switch (status) {
+    case "starting":
+      return template.build_time_stats.start
+    case "stopping":
+      return template.build_time_stats.stop
+    case "deleting":
+      return template.build_time_stats.delete
+    default:
+      return undefined
+  }
+}
+
 const estimateFinish = (
   startedAt: Dayjs,
-  buildEstimate: number,
-): [number, string] => {
-  const realPercentage = dayjs().diff(startedAt) / buildEstimate
+  median: number,
+  stddev: number,
+): [number | undefined, string] => {
+  const sinceStart = dayjs().diff(startedAt)
+  const secondsLeft = (est: number) =>
+    Math.max(
+      Math.ceil(dayjs.duration((1 - sinceStart / est) * est).asSeconds()),
+      0,
+    )
 
-  const maxPercentage = 1
-  if (realPercentage > maxPercentage) {
-    return [maxPercentage * 100, "Any moment now..."]
-  }
+  const lowGuess = secondsLeft(median)
+  const highGuess = secondsLeft(median + stddev)
 
-  return [
-    realPercentage * 100,
-    `~${Math.ceil(
-      dayjs.duration((1 - realPercentage) * buildEstimate).asSeconds(),
-    )} seconds remaining...`,
+  const anyMomentNow: [number | undefined, string] = [
+    undefined,
+    "Any moment now...",
   ]
+
+  // If variation is too high (and greater than second), don't show
+  // progress bar and give range.
+  const highVariation = stddev / median > 0.1 && highGuess - lowGuess > 1
```

Comment:
I might do this logic on the backend instead and return simpler API values. In the current format, it's not possible for our CLI to display a progress bar or estimated time remaining without duplicating this logic.